### ISSUE TITLE
why the data-type of output is quint8 in static quantize? what  static quantize does under the hood?

### ISSUE THREAD

**Issue by litaozijin** [pytorch/pytorch#28771]
Here is a example of static quantize,My python is version 3.7 and torch is 1.3.:
`
import torch
import torch.nn as nn
m = nn.quantized.Linear(20,30)
input = torch.randn(128.20)
input = torch.quantize_per_tensor(input,1.0,0,torch.quint8)
output = m(input)
print (output.dtype)
`
I feel confused why the data-type of output is quint8 rather than float or int-32 in static quantize，it may  will cause accuracy loss when model 'm' is joint with a  soft-max layer at last ?what static quantize does underlying?

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a @pytorch/quantization

**Comment by jianyuh** [comment]
The data type of the output is quint8, and we have the accumulation type as int32 (which can grant enough accuracy during the quantization).

**Comment by litaozijin** [comment]
> The data type of the output is quint8, and we have the accumulation type as int32 (which can grant enough accuracy during the quantization).

Of course , accumulating quint8 into int32  grant enough accuracy during quantization.But i wonder whether the output of quint8  will cause accuracy loss when it's passed to next accuracy-sensitive layer ,such as softmax or sigmod layer?
Otherwise,how accumulation of int32  is requantized to quint8 during quantization?

**Comment by lly-zero-one** [comment]
For each op in static quantization, the activation input and output are always in int8 type. I think this is the contract of int8 quantization, in which each value is mapped into one of the 256 levels. Yes, if the op is accuracy-sensitive, it may cause accuracy loss. That's why we have quantization aware training to mitigate this kind of issue and make your model accuracy be robust to quantization.

**Comment by litaozijin** [comment]
> For each op in static quantization, the activation input and output are always in int8 type. I think this is the contract of int8 quantization, in which each value is mapped into one of the 256 levels. Yes, if the op is accuracy-sensitive, it may cause accuracy loss. That's why we have quantization aware training to mitigate this kind of issue and make your model accuracy be robust to quantization.

Is there a function whose input and weights are quantized int8 and output fp32 for linear and conv2d?

**Comment by lly-zero-one** [comment]
> > For each op in static quantization, the activation input and output are always in int8 type. I think this is the contract of int8 quantization, in which each value is mapped into one of the 256 levels. Yes, if the op is accuracy-sensitive, it may cause accuracy loss. That's why we have quantization aware training to mitigate this kind of issue and make your model accuracy be robust to quantization.
> 
> Is there a function whose input and weights are quantized int8 and output fp32 for linear and conv2d?

If you really want to do that, you can manually insert dequant to convert the data to non-quantized domain.
